aeron-client/src/main/java/io/aeron/ChannelUri.java

```diff
@@ -507,6 +511,28 @@ public static long getTag(final String paramValue)
             AsciiEncoding.parseLongAscii(paramValue, 4, paramValue.length() - 4) : INVALID_TAG;
     }
 
+    /**
+     * Create a channel URI for a destination, i.e. a channel that uses {@code media} and {@code interface} parameters
+     * of the original channel and adds specified {@code endpoint} to it. For example given the input channel is
+     * {@code aeron:udp?mtu=1440|ttl=0|endpoint=localhost:8090|term-length=128k|interface=eth0} and the endpoint is
+     * {@code 192.168.0.14} the output of this method will be {@code aeron:udp?endpoint=192.168.0.14|interface=eth0}.
+     *
+     * @param channel  for which the destination is being added.
+     * @param endpoint for the target destination.
+     * @return new channel URI for a destination.
+     */
+    public static String createDestinationUri(final String channel, final String endpoint)
```

Comment:
Would this code be cleaner if we used the ChannelUriStringBuilder?

Reply:
Yes, it would but that would mean having a circular dependency between `ChannelUriStringBuilder` and `ChannelUri`. If `createDestinationUri` is moved elsewhere (maybe even to the `ChannelUriStringBuilder`) we could use the `ChannelUriStringBuilder`.

Reply:
Fair point I missed that.